### DISCUSSION TITLE
Beautiful Maps with R (V): Point densities | One world

### DISCUSSION THREAD

**Discussion by giscus** [dieghernan/dieghernan.github.io#32] [Blog]
# Beautiful Maps with R (V): Point densities | One world

Bertin's dot density maps with R and GHSL - Another way to represent demographics on a map.

https://dieghernan.github.io/202312_bertin_dots/

**Comment by dieghernan** [comment]
Answering to a previous comment (pre-migration to [giscus comments](https://giscus.app/es):

> _By. hyunchu,kim, 2023-12-19T07:28:04Z_
>
> sorry. i can't follow this code:
> `pop_init <- rast(global_tiff_path, win = as_spatvector(owin))`
> `[window<-] 'value' should be a SpatExtent, NULL or NA`


So `pop_init <- rast(global_tiff_path, win = as_spatvector(owin))` reads a raster but only in the region delimited by the object passed in `win`. As raster files tend to be quite large this saves a lot of resources by limiting the load process to a specific zone.

Maybe you have skipped previous steps (i.e. the creation of `owin` or failed to load tidyterra or terra)? I created a minimal example and its work fine:

``` r
library(terra)
#> terra 1.7.65
library(giscoR)
library(sf)
library(dplyr)
library(ggplot2)
library(tidyterra)

# Create an owin
owin <- gisco_get_countries(country = "LU", spatialtype = "LB") %>%
  st_buffer(10000) %>%
  st_transform(4326)

# Read with owin, will zoom to where we expect
f <- system.file("ex/elev.tif", package = "terra")

r <- rast(f, win = tidyterra::as_spatvector(owin))

autoplot(r) +
  geom_sf(data = owin, fill = NA)
```

![](https://i.imgur.com/SEcSmql.png)

``` r

# no owin

r2 <- rast(f)

autoplot(r2) +
  geom_sf(data = owin, fill = NA)
```

![](https://i.imgur.com/kuyvZ0M.png)

<sup>Created on 2024-01-24 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>
